docs/scos/dev/upgradability-services/new-documentation/unique-entity-name-not-unique.md

```diff
@@ -0,0 +1,296 @@
+---
+title: Entity name is not unique
+description: Reference information for evaluator and upgrader tools.
+last_updated: Mar 23, 2022
+template: concept-topic-template
+---
+
+## Entity name is not unique
+
+The names of the following entities should be unique:
+
+* Transfers
+* Transfer properties
+* Database tables
+* Database columns
+* Methods
+* Constants
+
+If a minor or major release introduces an entity with the same name, the entity in your project might change behavior or cause issues.
+
+## Making transfer names unique
+
+To resolve the errors provided in the preceding examples, rename the transfer names. For example, add the project name as a prefix.
+
+{% info_block infoBox "Future-proof names" %}
+
+The names should be unique to the extent of making it impossible to accidentally match the name of a core transfer introduced in the future.
+
+{% endinfo_block %}
+
+#### Examples of not unique transfer name
+
+```xml
+<?xml version="1.0"?>
+<transfers xmlns="spryker:transfer-01" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="spryker:transfer-01 http://static.spryker.com/transfer-01.xsd">
+    <transfer name="ProductAbstractStore">
+        <property name="productAbstractSku" type="string"/>
+        <property name="storeName" type="string"/>
+    </transfer>
+</transfers>
+```
+
+#### Examples related error in the Evaluator output for transfer:
+```bash
+---------------- ----------------------------------------------------------------------------------------------------
+NotUnique:TransferName Transfer object name ProductAbstractStore has to have project prefix Pyz in /0000-fork-b2c-demo-shop/src/Pyz/Shared/Product/Transfer/product.transfer.xml, like PyzProductAbstractStore
+---------------------- ----------------------------------------------------------------------------------------------------
+```
+
+#### Examples of unique transfer name
+
+Renamed transfer name:
+
+```xml
+<?xml version="1.0"?>
+<transfers xmlns="spryker:transfer-01" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="spryker:transfer-01 http://static.spryker.com/transfer-01.xsd">
+    <transfer name="PyzProductAbstractStore">
```

Comment:
We need to point that `Pyz` as an example project prefix and it can be a custom one.
@VladislavStrelchenko, can you please provide more information? Thank you.

Reply:
Yes, we a close to releasing the next feature: the evaluator will be able to work with a list of custom project namespaces [Pyz, ..., ...]
We will be able to configure it in the file tooling.yaml - more detail will be after review and doc writing, probably this week

Reply:
@VladislavStrelchenko please add more information to this doc afterwards